fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/rescheduleloan/service/LoanRescheduleRequestWritePlatformServiceImpl.java

```diff
@@ -195,29 +196,24 @@ public CommandProcessingResult create(JsonCommand jsonCommand) {
                 submittedOnDate = jsonCommand.localDateValueOfParameterNamed(RescheduleLoansApiConstants.submittedOnDateParamName);
             }
 
-            // initially set the value to null
-            LocalDate rescheduleFromDate = null;
-
             // start point of the rescheduling exercise
             Integer rescheduleFromInstallment = null;
 
             // initially set the value to null
             LocalDate adjustedDueDate = null;
 
+            LocalDate rescheduleFromDate = jsonCommand
+                    .localDateValueOfParameterNamed(RescheduleLoansApiConstants.rescheduleFromDateParamName);
             // check if the parameter is in the JsonCommand object
-            if (jsonCommand.hasParameter(RescheduleLoansApiConstants.rescheduleFromDateParamName)) {
-                // create a LocalDate object from the "rescheduleFromDate" Date
-                // string
-                LocalDate localDate = jsonCommand.localDateValueOfParameterNamed(RescheduleLoansApiConstants.rescheduleFromDateParamName);
-
-                if (localDate != null) {
-                    // get installment by due date
-                    LoanRepaymentScheduleInstallment installment = loan.getRepaymentScheduleInstallment(localDate);
-                    rescheduleFromInstallment = installment.getInstallmentNumber();
-
-                    // update the value of the "rescheduleFromDate" variable
-                    rescheduleFromDate = localDate;
-                }
+            if (rescheduleFromDate != null) {
+                final boolean isProgressiveLoanInterestRateChangeOperation = jsonCommand
+                        .hasParameterValue(RescheduleLoansApiConstants.newInterestRateParamName)
+                        && loan.getLoanProductRelatedDetail().getLoanScheduleType() == LoanScheduleType.PROGRESSIVE;
+                // get installment by due date
+                LoanRepaymentScheduleInstallment installment = isProgressiveLoanInterestRateChangeOperation
```

Comment:
I reckon we can use `loan.getRelatedRepaymentScheduleInstallment` for both case, no?

Reply:
we have to ensure if we do due date rescheduling than the provided date should be on due date.
We can use this method also, but in that case we have to check the provided date is an due date.

Reply:
I reckon we already doing it within validations, other was installment is null and NPE will be triggered the next line

Reply:
I totally lost in this GitHub code, I just see this is not the Validator class.
This is historical here, because originally the `getRelatedRepaymentScheduleInstallment` method excluded the due date, but the updated logic already accepts that. So you are right, here we can just use that method.